R/radqlqc.R

```diff
@@ -0,0 +1,1304 @@
+      QSTESTCD == "EOR0110" ~ "QS02810",
+      QSTESTCD == "EOR0111" ~ "QS02811",
+      QSTESTCD == "EOR0112" ~ "QS02812",
+      QSTESTCD == "EOR0113" ~ "QS02813",
+      QSTESTCD == "EOR0114" ~ "QS02814",
+      QSTESTCD == "EOR0115" ~ "QS02815",
+      QSTESTCD == "EOR0116" ~ "QS02816",
+      QSTESTCD == "EOR0117" ~ "QS02817",
+      QSTESTCD == "EOR0118" ~ "QS02818",
+      QSTESTCD == "EOR0119" ~ "QS02819",
+      QSTESTCD == "EOR0120" ~ "QS02820",
+      QSTESTCD == "EOR0121" ~ "QS02821",
+      QSTESTCD == "EOR0122" ~ "QS02822",
+      QSTESTCD == "EOR0123" ~ "QS02823",
+      QSTESTCD == "EOR0124" ~ "QS02824",
+      QSTESTCD == "EOR0125" ~ "QS02825",
+      QSTESTCD == "EOR0126" ~ "QS02826",
+      QSTESTCD == "EOR0127" ~ "QS02827",
+      QSTESTCD == "EOR0128" ~ "QS02828",
+      QSTESTCD == "EOR0129" ~ "QS02829",
+      QSTESTCD == "EOR0130" ~ "QS02830",
+      TRUE ~ QSTESTCD
+    )
+  )
+
+  adqlqc1 <- left_join(
+    adqlqc,
+    gdsr_param_adqlqc
+  )
+
+  return(adqlqc1)
+}
+
+
+
+#' Scale calculation for ADQLQC data
+#'
+#' @param adqlqc1 prepared data generated from the prep_adqlqc function
+#'
+#' @return dataframe
+#'
+#' @import dplyr
+#' @importFrom stringr str_extract str_detect
+#' @importFrom rgdsr adamv_parameters
+#'
+#'
+#' @examples
+#' \dontrun{
+#' df <- calc_scales(adqlqc1)
+#' }
+calc_scales <- function(adqlqc1) {
+  # Prep scale data ---------------------------------------------------------
+  # parcat2 = scales or global health status
+  # global health status/scales data
+
+  # EORTC QLQ-C30 V3.0
+  eortc_qlq_c30 <- get_questionnaire_sdtm(qscat = "EORTC QLQ-C30 V3.0")
+
+  # QSTESTCD: EOR0131 to EOR0145 (global health status and scales)
+  eortc_qlq_c30_sub <- filter(
+    eortc_qlq_c30,
+    !(as.numeric(str_extract(.data$QSTESTCD, "\\d+$")) >= 101 &
+      as.numeric(str_extract(.data$QSTESTCD, "\\d+$")) <= 130)
+  ) %>%
+    mutate(
+      PARAMCD = case_when(
+        .data$QSTESTCD == "EOR0131" ~ "QS028QL2",
+        .data$QSTESTCD == "EOR0132" ~ "QS028PF2",
+        .data$QSTESTCD == "EOR0133" ~ "QS028RF2",
+        .data$QSTESTCD == "EOR0134" ~ "QS028EF",
+        .data$QSTESTCD == "EOR0135" ~ "QS028CF",
+        .data$QSTESTCD == "EOR0136" ~ "QS028SF",
+        .data$QSTESTCD == "EOR0137" ~ "QS028FA",
+        .data$QSTESTCD == "EOR0138" ~ "QS028NV",
+        .data$QSTESTCD == "EOR0139" ~ "QS028PA",
+        .data$QSTESTCD == "EOR0140" ~ "QS028DY",
+        .data$QSTESTCD == "EOR0141" ~ "QS028SL",
+        .data$QSTESTCD == "EOR0142" ~ "QS028AP",
+        .data$QSTESTCD == "EOR0143" ~ "QS028CO",
+        .data$QSTESTCD == "EOR0144" ~ "QS028DI",
+        .data$QSTESTCD == "EOR0145" ~ "QS028FI",
+        TRUE ~ .data$QSTESTCD
+      )
+    ) %>%
+    select(-.data$publication_name)
+
+  # ADaM global health status and scales from gdsr
+  gdsr_param_adqlqc <- adamv_parameters()[["ADQLQC"]] %>%
+    select(
+      .data$PARAM,
+      .data$PARAMCD,
+      .data$PARCAT2,
+      .data$PARCAT1N
```

Comment:
Can we extract the dataframe from rgdsr, and just keep these columns and save it as dataset, and load it in?

This is repeated multiple times. Note, we are not updating this, so any version is fine

Reply:
Thank you, functions and code for rgdsr removed. Generated data stored in inst/data.